### PR TITLE
Google popup by feroxes

### PR DESCRIPTION
**Pull request by feroxes**
### Google Popup

[Demo](https://feroxes.github.io/big_brother_popup/) |
[Code base](https://github.com/feroxes/frontend-2019-homeworks/tree/google-popup/submissions/feroxes/google-popup)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/feroxes/google-popup/style.css [hunk 1]
```diff
@@ -0,0 +1,207 @@
+header {
```
Comment (A-Ostrovnyy) [inline submissions/feroxes/google-popup/style.css]: Avoid styling by tag name.

File: submissions/feroxes/google-popup/style.css [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+.menu a:visited {
+    color: #000;
+}
+
+.menu a:hover {
+    color: #000;
+    text-decoration: underline;
+}
+
+#dots_label {
```
Comment (A-Ostrovnyy) [inline submissions/feroxes/google-popup/style.css]: Avoid styling by `#id`.

File: submissions/feroxes/google-popup/style.css [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    display: flex;
+}
+
+#more:checked ~ #more-icons {
+    display: none;
+}
+
+#more:checked ~ .hidden {
+    display: flex;
+}
```
Comment (A-Ostrovnyy) [inline submissions/feroxes/google-popup/style.css]: Add empty line in the end of every file.

**Comment by feroxes** [comment]
@A-Ostrovnyy Please explain, why it is better to avoid styling by tag name and id? Thanks)

File: submissions/feroxes/google-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    <title>Google</title>
+</head>
+<body>
+    <header>
+        <div class="menu">
+            <a href="#">Gmail</a>
+            <a href="#">Images</a>
+            <label id="dots_label" for="dots"></label>
+            <input id="dots" type="checkbox">
+            <div class="google_services">
```
Comment (A-Ostrovnyy) [inline submissions/feroxes/google-popup/index.html]: For lists of items better use `ul > li` or `dl > dt` tags. It's better for accessibility. Then screen readers understand list block as structured information block.

**Comment by Shramkoweb** [comment]
> @A-Ostrovnyy Please explain, why it is better to avoid styling by tag name and id? Thanks)

More info about why is batter to avoid styling by id [LINK](https://dev.to/claireparker/reasons-not-to-use-ids-in-css-4ni4)

**Comment by feroxes** [comment]
> Good work.
> But your menu icon and other in this row did not accessible from keyboard. Please fix this.

@A-Ostrovnyy Please, explain how does it work? I cover every element in header by li tag. When I press Tab, I see blue outline on first and then on second li, but other li`s doesn't have it.

**Comment by A-Ostrovnyy** [comment]
@feroxes When i just download your page(popup is closed) I press Tab key and menu items must иу illuminated by outline. In your case first two items(Gmail and Images) illuminated and then focus is missing. Other menu items(menu, bell and avatar icons) must be illuminated by outline too.

**Comment by feroxes** [comment]
Ok. I just push my changes.
I cover every header tag by 'li' and it doesn't work. Only first and second tags has outline.

File: submissions/feroxes/google-popup/index.html [hunk 5]
```diff
@@ -10,12 +10,12 @@
...
-        <div class="menu">
-            <a href="#">Gmail</a>
-            <a href="#">Images</a>
-            <label id="dots_label" for="dots"></label>
-            <input id="dots" type="checkbox">
+    <div class="header">
+        <ul class="menu">
+            <li><a href="#">Gmail</a></li>
+            <li><a href="#">Images</a></li>
+            <li><label id="dots_label" class="dots_label" for="dots"></label></li>
```
Comment (A-Ostrovnyy) [inline submissions/feroxes/google-popup/index.html]: In your case better change markup for input:
```
<label class="checkbox-label-wrapper">
        <input type="checkbox"
               name="input"
               id="input1"
               class="visually-hidden input">
        <span class="checkbox-label">
        </span>
    </label>
```
and CSS selector for accessibility:
```
.input:focus + .checkbox-label
```

Comment (feroxes) [reply submissions/feroxes/google-popup/index.html]:
```
.dots_input:checked ~ .google_services {
    display: flex;
}
```
This code wouldn't work if I put input into the label.

Comment (A-Ostrovnyy) [reply submissions/feroxes/google-popup/index.html]: My bad forget about items wrapper.
Here is [demo](https://jsfiddle.net/0aw4Lf36/)  with simple realisation. Hope it helps you.

**Comment by feroxes** [comment]
Hi!)
I couldn't solve the problem, as you said. But corrected it with the addition of the attribute tabindex:)

**Comment by A-Ostrovnyy** [comment]
@feroxes I spent some time to figure out with your 'problem'. The main problem is markup, correct markup you can see [here](https://gist.github.com/A-Ostrovnyy/d638d14afe380c0756560f2be93471af).
Second problem is hiding checkbox. You hide it by `display: none;` this is a wrong way. Use class `.visually-hidden` from my demo from slides.
If you have some problem, feel free to ask me in telegram into russian or ukrainian.

**Comment by feroxes** [comment]
All done:)

**Comment by A-Ostrovnyy** [comment]
Good job!
Avatar icon and button More(inside popup) not accessible from the keyboard.

**Pull request merged**